### PR TITLE
docs(docs): add docs on site theme, theme branding, and more

### PR DESCRIPTION
**Pull request by jameswburke**
## Issue(s): Relates to or closes...
None.

## Summary: This pull request will...
Add docs on Site Theme, Theme Options, and other misc additions.

## Tests: I know this code works because...
This PR has been pushed to http://storybook-qa.irvingjs.com/

### CODE REVIEW

File: stories/07-site-theme/01-site-theme-provider.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+#### Example `/container/themes/default.js`
+```
+import styled from 'styled-components';
+import { siteTheme } from '@irvingjs/styled/utils';
+
+/* eslint-disable import/prefer-default-export */
+export const ContainerWrapper = styled.div`
+  border: 1px solid ${siteTheme('colors.brand.secondary')};
+  background-color: ${siteTheme('header.background_color')};
```
Comment (amyevans) [inline stories/07-site-theme/01-site-theme-provider.stories.mdx:121]: If we could sneak in a `rem` example here too I think that'd be a good win :)

**Review by amyevans — APPROVED** [review]
✨

File: stories/03-features/Authentication.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<Meta title="Features|Authentication" />
+
+# Authentication
+
+## Basic Authentication
+Basic auth can be enabled using environment variables. They will automatically apply the Authorization header for requests to the app. This will prompt the user to enter the valid credentials to access the site.
+
+* `BASIC_AUTH` - Put your entire site behind basic auth when running in production mode.
+* `BASIC_AUTH_USERNAME` - Basic auth username. This will be used for both sitewide basic auth and cache endpoints.
+* `BASIC_AUTH_PASSWORD` - Basic auth password. This will be used for both sitewide basic auth and cache endpoints.
```
[the remark this answers is not in the document]
Comment (jameswburke) [reply stories/03-features/Authentication.stories.mdx:10]: Agreed, I'm fuzzy on those details, or I'd update myself.

File: stories/05-templating/01-Inline-Styles.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+Site Theme:
+```json
+{
+	"colors": {
+		"primary": "#735fea"
+	},
+	"header": {
+		"background_color": "colors.brand.primary"
```
[the remark this answers is not in the document]
Comment (jameswburke) [reply stories/05-templating/01-Inline-Styles.stories.mdx]: I tweaked some of this, not sure I hit exactly what you're looking for, but we can adjust as needed.

**Pull request merged**